src/settings.ts

```diff
@@ -48,59 +60,73 @@ export function addCalendarButton(
 	return new Setting(containerEl)
 		.setName("Calendars")
 		.setDesc("Add calendar")
-		.addDropdown((d) =>
-			dropdown = d
-			.addOptions({
-				local: "Local",
-				icloud: "iCloud",
-				caldav: "CalDAV",
-				ical: "Remote (.ics format)",
-			})
+		.addDropdown(
+			(d) =>
+				(dropdown = d.addOptions({
+					local: "Local",
+					icloud: "iCloud",
+					caldav: "CalDAV",
+					ical: "Remote (.ics format)",
+				}))
 		)
 		.addExtraButton((button) => {
 			button.setTooltip("Add Calendar");
 			button.setIcon("plus-with-circle");
 			button.onClick(() => {
-				let modal = new ReactModal(
-					app,
-					async () => {
-						await plugin.loadSettings();
-						const usedDirectories = (listUsedDirectories ? listUsedDirectories : () =>
-							plugin.settings.calendarSources
-								.map((s) => s.type === "local" && s.directory)
-								.filter((s): s is string => !!s)
-						)();
-
-						return createElement(AddCalendarSource, {
-							source: makeDefaultPartialCalendarSource(
-								dropdown.getValue() as CalendarSource["type"]
-							),
-							directories: directories.filter(
-								(dir) => usedDirectories.indexOf(dir) === -1
-							),
-							submit: async (source: CalendarSource) => {
-								if (source.type === "caldav" || source.type === "icloud") {
-									let sources = await new RemoteSource(source).importCalendars();
-									if (sources instanceof FCError) {
-										new Notice(sources.message);
-									} else {
-										sources.forEach((source) => submitCallback(source));
-									}
+				let modal = new ReactModal(app, async () => {
+					await plugin.loadSettings();
+					const usedDirectories = (
+						listUsedDirectories
+							? listUsedDirectories
+							: () =>
+									plugin.settings.calendarSources
+										.map(
+											(s) =>
+												s.type === "local" &&
+												s.directory
+										)
+										.filter((s): s is string => !!s)
+					)();
+
+					return createElement(AddCalendarSource, {
```

Comment:
```suggestion
					return React.createElement(AddCalendarSource, {
```

With the above suggestion for imports.